src/main/java/com/actiontech/dble/plan/optimizer/MyOptimizer.java

```diff
@@ -38,11 +38,17 @@ public static PlanNode optimize(PlanNode node, @Nullable HintPlanInfo hintPlanIn
         TraceManager.TraceObject traceObject = TraceManager.threadTrace("optimize-for-sql");
         TraceManager.log(ImmutableMap.of("plan-node", node), traceObject);
         try {
-            // PreProcessor SubQuery ,transform in sub query to join
-            if (SystemConfig.getInstance().isInSubQueryTransformToJoin()) {
-                node = SubQueryPreProcessor.optimize(node);
+
+            if (hintPlanInfo == null || !hintPlanInfo.isIn2join()) {
```

Comment:
hintPlanInfo is nonNull. use hintPlanInfo.isEmpty()

Reply:
hintPlanInfo.isEmpty() deviates from the actual meaning